modules/charon-core/src/test/java/org/wso2/charon3/core/utils/AttributeUtilTest.java

```diff
@@ -39,6 +50,121 @@
+                {10.0, DECIMAL, Double.class},
+                {"2021-04-20T09:06:19.839Z", DATE_TIME, Instant.class},
+                {0b010110, BINARY, Byte.class},
+                {"referenceString", REFERENCE, String.class},
+                {"complexString", COMPLEX, String.class},
+                {null, STRING, null}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetAttributeValueFromString")
+    public void testGetAttributeValueFromString(Object attributeValue, SCIMDefinitions.DataType
+            dataType, Object expectedAttributeValueTypeFromString) throws CharonException, BadRequestException {
+
+        Object attributeValueFromString = AttributeUtil.getAttributeValueFromString(attributeValue, dataType);
+
+        if (attributeValue != null) {
+            Assert.assertEquals(attributeValueFromString.getClass(), expectedAttributeValueTypeFromString);
+        } else {
+            Assert.assertNull(attributeValueFromString);
+        }
+    }
+
+    @DataProvider(name = "dataForGetStringValueOfAttribute")
+    public Object[][] dataToGetStringValueOfAttribute() {
+
+        Instant instant = Instant.parse("2021-04-20T09:06:19.839Z");
+
+        return new Object[][]{
+
+                {"username", STRING, "username"},
+                {true, BOOLEAN, "true"},
+                {2, INTEGER, "2"},
+                {10.0, DECIMAL, "10.0"},
+                {instant, DATE_TIME, "2021-04-20T09:06:19.839Z"},
+                {0b010110, BINARY, "22"},
+                {"referenceString", REFERENCE, "referenceString"},
+                {"complexString", COMPLEX, "complexString"}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetStringValueOfAttribute")
+    public void testGetStringValueOfAttribute(Object attributeValue, SCIMDefinitions.DataType
+            dataType, Object expectedStringValueOfAttribute) throws CharonException {
+
+        Object stringValueOfAttribute = AttributeUtil.getStringValueOfAttribute(attributeValue, dataType);
+        Assert.assertEquals(stringValueOfAttribute, expectedStringValueOfAttribute);
+    }
+
+    @DataProvider(name = "dataForParseDateTime")
+    public Object[][] dataToParseDateTime() {
+
+        Instant instant = Instant.parse("2021-04-20T09:06:19.839Z");
+
+        return new Object[][]{
+
+                {"2021-04-20T09:06:19.839Z", instant, "SUCCESS"},
+                {"", null, "SUCCESS"},
+                {"2021-04-20T09", null, "DATE_TIME_EXCEPTION"}
+        };
+    }
+
+    @Test(dataProvider = "dataForParseDateTime")
+    public void testParseDateTime(String dateTimeString, Instant expectedLocalDateTime, String expectedResult)
+            throws CharonException {
+
+        String result = "";
+        try {
+            Instant localDateTime = AttributeUtil.parseDateTime(dateTimeString);
+            Assert.assertEquals(localDateTime, expectedLocalDateTime);
+            result = "SUCCESS";
+        } catch (CharonException e) {
```

Comment:
Since we catch the CharonException here, we don't need to throw it as well.

Reply:
fixed in 3149d105c7620f5b32618b35eb6e7cc3e426784c